exec/vector/src/main/java/org/apache/drill/exec/vector/accessor/reader/UnionReaderImpl.java

```diff
@@ -182,6 +182,15 @@ public void reposition() {
     }
   }
 
+  @Override
+  public void rebind() {
+    for (int i = 0; i < variants.length; i++) {
```

Comment:
This loop also may be rewritten with a for-each loop, or even with a stream.

Reply:
Also performance-sensitive code.